packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -431,6 +431,19 @@ export interface BucketAttributes {
   readonly region?: string;
 }
 
+/**
+ * Props common to all buckets
+ */
+export interface BucketBaseProps extends ResourceProps {
+
+  /**
+   * The role to be used by the notifications handler
+   *
+   * @default - a new role will be created.
+   */
+  readonly notificationsHandlerRole?: iam.IRole;
```

Comment:
This needs to be available for imported buckets as well since they support bucket notifications too.